qemu/tests/in_place_upgrade_legacy.py

```diff
@@ -0,0 +1,126 @@
+import re
+import logging
+import aexpect
+
+
+from virttest import error_context
+from qemu.tests.in_place_upgrade import IpuTest
```

Comment:
You should not import a module in a case from another case, if it has common functions, save it into `provider` directory.

Reply:
Updated, and move the common functions to provide and feel free to review it when you are free, thanks